### ISSUE TITLE
Should check prefix on functions created with fset

### ISSUE THREAD

**Issue by riscy** [purcell/package-lint#156]
In a package named `foo`, the following does not raise any package-lints:
```elisp
(fset 'bar #'foo-bar)
```
whereas the following does:
```elisp
(defalias 'bar #'foo-bar)
```
i.e. `error: Aliases should start with the package's prefix "foo".`

These are effectively the same (per the docs, defalias normally just calls fset).

This popped up here: https://github.com/melpa/melpa/pull/6614

**Comment by purcell** [comment]
What happens when the symbol is locally-bound? If this mechanism can be used as a clunky alternative to `letf`/`flet` then I'd worry that a check might be too strict. Other than that, though, I think this is a very easy change to implement.

**Comment by alphapapa** [comment]
Checking `fset` calls like this won't raise a warning for calls like this, right?  https://github.com/alphapapa/org-ql/blob/master/org-ql.el#L341

**Comment by riscy** [comment]
> What happens when the symbol is locally-bound? If this mechanism can be used as a clunky alternative to letf/flet then I'd worry that a check might be too strict.

Oops, I hadn't even thought of local symbols (and the example @alphapapa linked).  You're right, I think it may be too strict.  I'll close this, but will keep an eye out for it going forward.

**Comment by purcell** [comment]
I mean, we could potentially check for `fset` at the top level, but that's a fuzzy concept, tbh, due to various legitimate top-level wrapping constructs like `let`, `eval-when-compile` etc.

**Comment by riscy** [comment]
Yeah -- when I saw it in the wild, it was wrapped in a `when`, so it'd be tough to nail down.  Ah well. :)

**Comment by purcell** [comment]
It's probably the only time I've seen that trick in a few thousand packages, so not a big deal, I guess!
